distribution/tools/keystore-cli/src/main/java/org/elasticsearch/common/settings/CreateKeyStoreCommand.java

```diff
@@ -21,41 +21,50 @@
 
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.Arrays;
 
 import joptsimple.OptionSet;
+import joptsimple.OptionSpec;
 import org.elasticsearch.cli.EnvironmentAwareCommand;
+import org.elasticsearch.cli.ExitCodes;
 import org.elasticsearch.cli.Terminal;
+import org.elasticsearch.cli.UserException;
 import org.elasticsearch.env.Environment;
 
 /**
- * A subcommand for the keystore cli to create a new keystore.
+ * A sub-command for the keystore cli to create a new keystore.
  */
 class CreateKeyStoreCommand extends EnvironmentAwareCommand {
 
+    private final OptionSpec<Void> passwordOption;
+
     CreateKeyStoreCommand() {
         super("Creates a new elasticsearch keystore");
+        this.passwordOption = parser.acceptsAll(Arrays.asList("p", "password"), "Prompt for password to encrypt the keystore");
     }
 
     @Override
     protected void execute(Terminal terminal, OptionSet options, Environment env) throws Exception {
-        Path keystoreFile = KeyStoreWrapper.keystorePath(env.configFile());
-        if (Files.exists(keystoreFile)) {
-            if (terminal.promptYesNo("An elasticsearch keystore already exists. Overwrite?", false) == false) {
-                terminal.println("Exiting without creating keystore.");
-                return;
+        SecureString password = null;
+        try {
+            Path keystoreFile = KeyStoreWrapper.keystorePath(env.configFile());
+            if (Files.exists(keystoreFile)) {
+                if (terminal.promptYesNo("An elasticsearch keystore already exists. Overwrite?", false) == false) {
+                    terminal.println("Exiting without creating keystore.");
+                    return;
+                }
+            }
+            KeyStoreWrapper keystore = KeyStoreWrapper.create();
+            password = options.has(passwordOption) ?
+                BaseKeyStoreCommand.readPassword(terminal, true) : new SecureString(new char[0]);
+            keystore.save(env.configFile(), password.getChars());
+            terminal.println("Created elasticsearch keystore in " + env.configFile());
```

Comment:
configFile is elasticsearch.yml. Shouldn't this be the location of the keystore?

Reply:
copy paste woes.